pyscriptjs/src/main.ts

```diff
@@ -234,6 +262,37 @@ export class PyScriptApp {
         logger.info('All paths fetched');
     }
 
+    /**
+     * Fetches all the python plugins specified in this.config, saves them on the FS and import
+     * them as modules, executing any plugin define the module scope
+     *
+     * @param runtime - runtime that will execute the plugins
+     */
+    async fetchPythonPlugins(runtime: Runtime) {
+        const plugins = this.config.plugins;
+        logger.info("Python plugins to fetch: ", plugins)
+        for (const singleFile of plugins) {
+            logger.info(`  fetching plugins: ${singleFile}`);
+            try {
+                const pathArr = singleFile.split('/');
+                const filename = pathArr.pop();
+                // TODO: Would be probably be better to store plugins somewhere like /plugins/python/
+                //       or similar
+                const destPath = `./${filename}`
+                await runtime.loadFromFile(destPath, singleFile);
+                const modulename = singleFile.replace(/^.*[\\/]/, '').replace('.py', '');
+
+                console.log(`importing ${modulename}`);
+                await runtime.run(`import ${modulename}`);
+
+            } catch (e) {
+                //Should we still export full error contents to console?
+                handleFetchError(<Error>e, singleFile);
+            }
+        }
+        logger.info("All plugins fetched");
+    }
+
```

Comment:
or maybe we could avoid/solve the code duplication by having some common function `fetchListOfURLs` which is called by both `fetch.ts` and `plugin.ts`

Reply:
I'm very -1 on this one. I think this would be a very bad UX from a plugins consumer standpoint. As a consumer I wouldn't want to care where pyscript put the plugins... I just want to specify where to load (~ fetch) them from.